modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -222,14 +222,12 @@ class ReplCalculator(
         // this is safe in recent spark versions, see SPARK-6414
         log.debug(s"Interrupting the cell: $killCellId")
         val jobGroupId = JobTracking.jobGroupId(killCellId)
-        try {
+        // make sure sparkContext is already available!
+        if (repl.interp.allDefinedNames.filter(_.toString == "sparkContext").nonEmpty) {
```

Comment:
- `sparkContext` is imported name, not a defined one ( see #492 ), so this check would always return `false` :)
- but when fixed with #492 `s"""sparkContext.cancelJobGroup("${jobGroupId}")"""` would cause kernel death if `init.sc` didn't complete fully.